leaderelection/leader_election.go

```diff
@@ -66,16 +78,33 @@ type leaderElection struct {
 }
 
 // NewLeaderElection returns the default & preferred leader election type
-func NewLeaderElection(clientset kubernetes.Interface, lockName string, runFunc func(ctx context.Context)) *leaderElection {
-	return NewLeaderElectionWithLeases(clientset, lockName, runFunc)
+// healthCheckTimeout determines the max duration beyond lease expiration
+// allowed before reporting unhealthy.
+func NewLeaderElection(
+	clientset kubernetes.Interface,
+	lockName string,
+	healthCheckTimeout time.Duration,
```

Comment:
Can we use one of the `With*` constructors so we don't need to modify this interface?

Reply:
or read the configuration from a file, if user need to change that, specify that in the file...

Reply:
Will update to a `With*` constructor. Adding a config file would be out of scope for this PR since a lot of flags could potentially be in a config file instead, though there's also a wider question of, what advantages does a ConfigMap offer over container flags